cluster-api/src/test/java/io/aiven/klaw/clusterapi/services/ConfluentCloudApiServiceTest.java

```diff
@@ -0,0 +1,286 @@
+    assertThat(resultMap.get("permission")).isEqualTo("ALLOW");
+  }
+
+  @Test
+  public void deleteAcls() throws Exception {
+    ClusterAclRequest clusterAclRequest = utilMethods.getConfluentCloudProducerAclRequest();
+    stubAcls();
+    when(restTemplate.exchange(
+            anyString(),
+            eq(HttpMethod.DELETE),
+            any(),
+            (ParameterizedTypeReference<ListAclsResponse>) any()))
+        .thenReturn(null);
+
+    String response = confluentCloudApiService.deleteAcls(clusterAclRequest);
+    assertThat(response).isEqualTo(ApiResultStatus.SUCCESS.value);
+  }
+
+  @Test
+  public void createTopic() throws Exception {
+    stubTopics();
+    ClusterTopicRequest clusterTopicRequest = utilMethods.getTopicRequest();
+    when(restTemplate.postForEntity(anyString(), any(), any())).thenReturn(null);
+
+    ApiResponse apiResponse = confluentCloudApiService.createTopic(clusterTopicRequest);
+    assertThat(apiResponse.getResult()).isEqualTo(ApiResultStatus.SUCCESS.value);
+  }
+
+  @Test
+  public void deleteTopic() throws Exception {
+    stubTopics();
+    ClusterTopicRequest clusterTopicRequest = utilMethods.getTopicRequest();
+    when(restTemplate.postForEntity(anyString(), any(), any())).thenReturn(null);
+
+    ApiResponse apiResponse = confluentCloudApiService.deleteTopic(clusterTopicRequest);
+    assertThat(apiResponse.getResult()).isEqualTo(ApiResultStatus.SUCCESS.value);
+  }
+
+  @Test
+  public void getTopicCreateObj() {
+    ClusterTopicRequest clusterTopicRequest = utilMethods.getTopicRequest();
+    TopicCreateRequest topicCreateRequest =
+        confluentCloudApiService.getTopicCreateObj(clusterTopicRequest);
+    assertThat(topicCreateRequest.getTopic_name()).isEqualTo(clusterTopicRequest.getTopicName());
+    assertThat(topicCreateRequest.getPartitions_count())
+        .isEqualTo(clusterTopicRequest.getPartitions());
+    assertThat(topicCreateRequest.getReplication_factor())
+        .isEqualTo(clusterTopicRequest.getReplicationFactor());
+    assertThat(topicCreateRequest.getConfigs())
+        .hasSize(clusterTopicRequest.getAdvancedTopicConfiguration().size());
+  }
+
+  @Test
+  public void getQueryParamsHost() {
+    ClusterAclRequest clusterAclRequest = utilMethods.getConfluentCloudProducerAclRequest();
+    String resourceType = "TOPIC";
+    String operation = "WRITE";
+    String response =
+        confluentCloudApiService.getQueryParams(
+            clusterAclRequest, resourceType, clusterAclRequest.getTopicName(), operation);
+    assertThat(response)
+        .isEqualTo(
+            "resource_type="
+                + resourceType
+                + "&resource_name="
+                + clusterAclRequest.getTopicName()
+                + "&pattern_type=LITERAL&principal=User:*&host="
+                + clusterAclRequest.getAclIp()
+                + "&operation="
+                + operation
+                + "&permission=ALLOW");
+  }
```

Comment:
I think you can use UriBuilder to build these if you want it will handle the & and ? .
But you may actually need more flexibility in the tests I'm not sure if it requires you to set a base e.g. https://klaw-project.io/

Reply:
Nice one. done.